### ISSUE TITLE
Documentation for Confirm since incorrect

### ISSUE THREAD

**Issue by caspChristian** [MicrosoftDocs/windows-powershell-docs#1628]
It says that Confirm defaults to false
But running this clearly asks for confirmation, so the default is true.
And to change this, you can't have -Confirm $false, you need -Confirm:$false
The only thing that explained this was the question at SO: https://stackoverflow.com/questions/8525572/powershell-colon-in-commandlet-parameters

This needs to be explained clearly, with examples, on all pages with this incorrect -Confirm switch exists

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 70649f0e-7299-1f0a-ad19-db9ed4141c22
* Version Independent ID: 96a9db1a-6cec-10e8-db25-64cc2a03dca4
* Content: [Remove-ADGroup (addsadministration)](https://docs.microsoft.com/en-us/powershell/module/addsadministration/remove-adgroup?view=win10-ps#feedback)
* Content Source: [docset/windows/addsadministration/Remove-ADGroup.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/live/docset/windows/addsadministration/Remove-ADGroup.md)
* Product: **w10**
* Technology: **powershell-windows**
* GitHub Login: @andreabarr
* Microsoft Alias: **v-anbarr**

**Comment by velkovb** [comment]
@caspChristian Hello Christian,

Thank you for trying to make MicrosoftDocs a better place.

By default switch parameters are always false. unless explicitly specified - [Switch Parameters](https://docs.microsoft.com/powershell/developer/cmdlet/types-of-cmdlet-parameters#switch-parameters). Here you are not prompted because of the -Confirm parameter but because the impact of the cmdlet is High. You could read more about cmdlet confirm impact and preference - [$ConfirmPreference](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_preference_variables#$ConfirmPreference), [ConfirmImpact](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_functions_cmdletbindingattribute#confirmimpact)

Furthermore, you could check the following article to understand confirmation requests - https://docs.microsoft.com/powershell/developer/cmdlet/users-requesting-confirmation

Let me know if you have any questions.

**Comment by caspChristian** [comment]
Well my tests prove it to be otherwise;
if -Confirm would be false then I would not need to give -Confirm as an argument at all.
If it is that -Confirm becomes false if given, than that should be clearly stated, however if that is the case than it is a horrible non-logical implementation. It would have been logical if the name was `-NoConfirm` which is what everyone is trying before finding this non local way of doing things.

Further more and again, all arguments are normally in the format `-Confirm $false` note the space, but this argument needs `-Confirm:$false` note colon, that should be documented, so that others don't have to search google for why they get syntax errors.

All of this could quite simple mitigated by just having clear and working examples of how you the confirmation is disabled (as it is, or rather was when I read it and posted this issue) I could only guess, and as I wrote, find the explanation on another site. Why wasn't the needed information on the documentation page to begin with!?

**Comment by velkovb** [comment]
Hello Christian,

I understand that you may find it confusing and it was for me as well in the beginning but if you read through the shared articles and understand switches and confirm preferences you will find the logic. This is how it is implemented and we cannot change that. This article may have slightly better explanation as well:
[Set-UserPhoto](https://docs.microsoft.com/powershell/module/exchange/mailboxes/set-userphoto?view=exchange-ps#parameters)

**Comment by caspChristian** [comment]
I should not need to read other pages to find documentation for illogical implementations.

Changing the behaviour is of course not an option, since it won't be back-ported tor older versions of PS anyway.
But that has never been my suggestion either.
Again for the 3rd time, the documentation should be clear, and even suggesting "read about these other commands" to try and just get something to quickly work is not a good, or even acceptable answer. Just fix the d***n documentation, Is it that hard!?

**Comment by velkovb** [comment]
Hello Christian,

This is how this functionality was implemented and we cannot change it. It may be illogical in your opinion but it is perfectly logical to the vast majority of people that have been using PowerShell for quite some time.

You cannot expect to learn everything about PowerShell from a single help page of a cmdlet, it requires time to get used to it like every scripting language and reading a lot more that a single help page. Most PowerShell courses have explanation about switch parameters and those that go deeper explain about the  $ConfirmPreference, ConfirmImpact. I have personally revisited those articles several times when I have had to write my own cmdlets supporting those arguments.
